java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/AdController.java

```diff
@@ -0,0 +1,276 @@
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAd;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadRequest;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdLoadResult;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueue;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueueListener;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdQueueManager;
+import com.chartboost.chartboostmediationsdk.ad.ChartboostMediationFullscreenAdShowListener;
+import com.chartboost.chartboostmediationsdk.domain.ChartboostMediationAdException;
+import com.chartboost.chartboostmediationsdk.domain.Keywords;
+import com.chartboost.core.ChartboostCore;
+import com.chartboost.core.initialization.ModuleInitializationResult;
+import com.chartboost.core.initialization.ModuleObserver;
+import com.chartboost.core.initialization.SdkConfiguration;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+public class AdController {
+
+    private final Map<String, ChartboostMediationFullscreenAd> fullscreenAds = new HashMap<>();
+    private final String TAG = "AdController";
+    private final Map<String, ChartboostMediationFullscreenAdQueue> adQueues = new HashMap<>();
+    private boolean shouldUseQueue = false;
+
+    public void initializeChartboost(
+            String appId,
+            OnLogStateChangeListener onLogStateChangeListener,
+            Context context,
+            MainActivity.OnInitializationCompleteListener onInitializationCompleteListener
+    ) {
+        onLogStateChangeListener.logState("Start initializing mediation");
+        ChartboostCore.initializeSdkFromJava(context, new SdkConfiguration(appId, new ArrayList<>(), new HashSet<>()), new ModuleObserver() {
+            @Override
+            public void onModuleInitializationCompleted(@NonNull ModuleInitializationResult moduleInitializationResult) {
+                if (moduleInitializationResult.getModuleId().equals(ChartboostMediationSdk.CORE_MODULE_ID)) {
+                    if (moduleInitializationResult.getException() == null) {
+                        onLogStateChangeListener.logState("Initialization finished");
+                        ChartboostMediationSdk.setTestMode(context, true);
+                        onInitializationCompleteListener.onInitializationCompleted();
+                    } else {
+                        onLogStateChangeListener.logState("Initialization exception %s");
+                        onLogStateChangeListener.logState(moduleInitializationResult.getException().getMessage());
+                    }
+                }
```

Comment:
I think we need an equivalent of `onInitializationCompleteListener()` for the `else` case (or one that accepts an error object for differentiation); otherwise, the calling app would never hear back.